docs/troubleshooting.md

```diff
@@ -0,0 +1,73 @@
+# Troubleshooting
+
+If you have problems, try to rebuild when compiling with `-DCMAKE_BUILD_TYPE=Debug`.
+In particular also rebuild the index.
+The release build assumes machine written words- and docsfiles and omits sanity checks for the sake of speed.
+
+## Run End-to-End Tests
+
+QLever includes a simple mechanism for testing the entire system from
+from building an index to running queries in a single command.
+
+In fact this End-to-End Test is run on Travis CI for every commit and Pull
+Request. It is also useful for local development however since it allows you to
+quickly test if something is horribly broken.
+
+Just like QLever itself the End-to-End Tests can be executed from a previously
+build QLever container
+
+**Note**: If you encounter permission issues e.g. if your UID is not 1000 or you
+are using docker with user namespaces, add the flag `-u root` to your `docker
+run` command or do a `chmod -R o+rw e2e_data`
```

Comment:
The same needs to be done for the `src/web` folder (as far as I know) as docker will otherwise potentially be unable to copy the index.html.

Reply:
The copying is done by the `docker` daemon which runs as root. The access is only a problem when the repository is checked out on an NFS where `root` can't read files owned by the normal user. That is a pretty obscure problem, though sadly it happens more often on our infrastructure as our homes are configured that way.